### PR TITLE
US05 FT Analytics: Add I18N for chart settings form

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**

### CODE REVIEW

File: analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/AnalyticsChartSetting.vue [hunk 1]
```diff
@@ -15,20 +15,20 @@
...
           v-model="tab"
           background-color="transparent"
           color="primary">
-          <v-tab>General</v-tab>
-          <v-tab>Colors</v-tab>
-          <v-tab>X axis</v-tab>
-          <v-tab>Y axis</v-tab>
-          <v-tab>Multiple charts</v-tab>
-          <v-tab>Data filters</v-tab>
+          <v-tab>{{ $t('analytics.settings.tab.General') }}</v-tab>
```
Comment (boubaker) [inline analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/AnalyticsChartSetting.vue]: It's better to always use simple keys for labels to be able to reuse them:
analytics.settings.tab.General => analytics.general
analytics.settings.tab.Colors => analytics.colors

Imagine you add in another place "Colors" label. Is it better to define a new key or reuse the existing key ?
In fact, it's better to reuse it to ensure that we don't increase the number of keys to translate for nothing.

(Same for other keys)

Comment (Rdidmohamedelhadi) [reply analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/AnalyticsChartSetting.vue]: OK I will do it

File: analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/SearchFilterForm.vue [hunk 2]
```diff
@@ -115,55 +115,57 @@ export default {
...
-      },
-    ],
-  }),
+  computed: {
+    searchFilterTypes() {
+      return [
+        {
+          text: this.$t('analytics.FilterTypes.equals'),
+          value: 'EQUAL',
+          placeholder: this.$t('analytics.FilterTypes.fieldValue.placeholder'),
```
Comment (boubaker) [inline analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/SearchFilterForm.vue]:
```suggestion
          placeholder: this.$t('analytics.fieldValuePlaceholder'),
```

Comment (boubaker) [reply analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/SearchFilterForm.vue]: same for other fields

File: analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/XAxisAggregationForm.vue [hunk 3]
```diff
@@ -18,7 +18,7 @@
                 :key="dateField.value"
                 :label="dateField.text"
                 :value="dateField.value" />
-              <v-radio label="Per custom field(s)" value="custom" />
+              <v-radio :label="$t('analytics.dateFields.custom')" value="custom" />
```
Comment (boubaker) [inline analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/XAxisAggregationForm.vue]:
```suggestion
              <v-radio :label="$t('analytics.perCustomField')" value="custom" />
```

File: analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/XAxisAggregationForm.vue [hunk 4]
```diff
@@ -30,7 +30,7 @@
                 <field-selection
                   v-model="item.field"
                   :fields-mappings="fieldsMappings"
-                  label="Field name"
+                  :label="$t('analytics.xAxis.aggregationField')"
```
Comment (boubaker) [inline analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/XAxisAggregationForm.vue]:
```suggestion
                  :label="$t('analytics.fieldName')"
```

File: analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/XAxisAggregationForm.vue [hunk 5]
```diff
@@ -113,6 +91,30 @@ export default {
     },
   }),
   computed: {
+    dateFields() {
+      return [
+        {
+          text: this.$t('analytics.dateFields.day'),
```
Comment (boubaker) [inline analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/XAxisAggregationForm.vue]:
```suggestion
          text: this.$t('analytics.perDay'),
```

Comment (boubaker) [reply analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/XAxisAggregationForm.vue]: Same for other fields

File: analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/YAxisAggregationForm.vue [hunk 6]
```diff
@@ -24,7 +24,7 @@
           <field-selection
             v-model="yAxisAggregation.field"
             :fields-mappings="fieldsMappings"
-            :placeholder="yAxisAggregationCardinality ? 'Distinct aggregation field' : 'Numeric aggregation field'"
+            :placeholder="yAxisAggregationCardinality ? $t('analytics.yAxis.aggregationCardinalityDistinct') : $t('analytics.yAxis.aggregationCardinalityNumeric')"
```
Comment (boubaker) [inline analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/YAxisAggregationForm.vue]:
```suggestion
            :placeholder="yAxisAggregationCardinality ? $t('analytics.distinctAggregationField') : $t('analytics.numericAggregationField')"
```

File: analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/YAxisAggregationForm.vue [hunk 7]
```diff
@@ -56,34 +56,36 @@ export default {
...
-        text: 'MIN of',
-        value: 'MIN',
-      },
-    ],
   }),
   computed: {
+    aggregationTypes() {
+      return [
+        {
+          text: this.$t('analytics.yAxis.aggregationTypes.count'),
```
Comment (boubaker) [inline analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/YAxisAggregationForm.vue]:
```suggestion
          text: this.$t('analytics.count'),
```

Comment (boubaker) [reply analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/YAxisAggregationForm.vue]: Same for other fields

File: analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/AnalyticsChartSetting.vue [hunk 8]
```diff
@@ -70,10 +70,10 @@
       <v-card-actions>
         <v-spacer />
         <button class="btn btn-primary ignore-vuetify-classes mr-1" @click="save">
-          Save settings
+          {{ $t('analytics.btn.save') }}
```
Comment (boubaker) [inline analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/AnalyticsChartSetting.vue]:
```suggestion
          {{ $t('analytics.save') }}
```

File: analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/AnalyticsChartSetting.vue [hunk 9]
```diff
@@ -70,10 +70,10 @@
...
         </button>
         <button class="btn ignore-vuetify-classes ml-1" @click="dialog = false">
-          Close
+          {{ $t('analytics.btn.close') }}
```
Comment (boubaker) [inline analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/AnalyticsChartSetting.vue]:
```suggestion
          {{ $t('analytics.close') }}
```

File: analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/SearchFilterForm.vue [hunk 10]
```diff
@@ -14,12 +14,12 @@
               <field-selection
                 v-model="item.field"
                 :fields-mappings="fieldsMappings"
-                placeholder="Field name" />
+                :placeholder="$t('analytics.FilterTypes.fieldName.placeholder')" />
```
Comment (boubaker) [inline analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/SearchFilterForm.vue]:
```suggestion
                :placeholder="$t('analytics.fieldName')" />
```

File: analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/SearchFilterForm.vue [hunk 11]
```diff
@@ -14,12 +14,12 @@
...
             </v-flex>
             <v-flex xs3 class="d-flex my-auto px-2">
               <select
                 v-model="item.type"
-                placeholder="Operator"
+                :placeholder="$t('analytics.FilterTypes.operator.placeholder')"
```
Comment (boubaker) [inline analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/SearchFilterForm.vue]:
```suggestion
                :placeholder="$t('analytics.operator')"
```

**Review by boubaker — APPROVED** [review]

**Pull request merged**